### ISSUE TITLE
[RFC] Call For Maintainers and Project Ownership

### ISSUE THREAD

**Issue by tannerlinsley** [react-static/react-static#1259]
# Call For Maintainers and Project Ownership

## Summary

As of today, both Nozzle and I (Tanner Linsley) will no longer be contributing to the project activily. The reasons for this change are motivated by both business, and personal reasons.

From Nozzle's perspective, Nozzle's core product offering and business model are not directly related to React Static, thus making contributions, dog-fooding, and availability more and more difficult to justify as the business grows. Nozzle has loved supported the project since its inception, but recognized that there is no better time for the project to move into more active hands.

I (Tanner Linsley), the creator and core-contributor of React Static, have chosen to continue persuing the future of javascript and react frameworks by investing more time into technologies like serverless prerendering, React SSR streaming, and micro-site generation. I'm going to start this journey by delving deeper into Next.js's latest serverless and hybrid export features.

At this point in time, I am actively looking for:

- A new company, commercial entity owner, or primary sponsor for the project. If you are interested, please email me at tannerlinsley@gmail.com!
- Core-contributors who would be willing to step up and manage the future of this fantastic library and also companies who would be willing to sponsor these developers in their efforts.

Depending on the outcome, React Static may remain a corporate sponsored project if a company does step forward, or at the very least will eventually become an open github organization with a few core-contributors added as admins. In either scenario, I am confident that there are developers out there who would be happy to participate in the future of React Static.

### Why are you moving on?

First off, thank you so much for using React Static. It's been an awesom adventure and I've learned so much from this library. This has been a very hard decision to make, but I believe it's for the best. Some of the main reasons for my moving on are:

- Serverless prerendering technology has gained my attention recently and I am really excited about the possibilities within it.
- My passion for static sites has ebbed and flowed over the years, but currently I am not using React Static as much as I had initially planned.

### Why do you plan on using Next.js?

Next.js has becoming extremely performant and mature over the years. When I initially examined it 2 years ago as a static site generator, it was very young. But today, Next.js is more than just an SSG. It's a full-fledged lamdba-powered serverless React Framework. It's technical advantages over static sites are very intriguing and I believe are the future of building websites in React. While static sites remain a very valid choice for smaller, less complex sites, I believe Next.js is the closest framework available that is well suited to solve the difficult problems related with shipping large or complex React websites on the internet.

### I use React Static... What should I do now?!

1. Don't jump to conclusions, worry, refactor, or do anything irrational! React Static will live on and you can keep using it as you always have.
2. If you plan on continuing to use React Static, consider contributing to it! There's not better time to start contributing to OSS than right now.
3. Give Next.js a try! You may be surprised at how much more easy and flexible it is compared to static site generators like React Static and Gatsby and even how cheap it is to operate still! Plus, there are a ton of other benefits to a lazily-server-rendered framework like faster content updates, smaller, faster builds, and even things like server-rendered user-content, A/B tests, etc.
4. Give Gatsby another try. It has become a very polished platform and will probably suit your needs if what you need is a very organized, all-inclusive static site generator. They even have a non-GraphQL mode now!

**Comment by sw-yx** [comment]
glad this is official. i've been helping triage issues but dont have time to take ownership given other priorities (i also lack a lot of knowledge, altho i know thats not a permanent barrier). would love for smarter people to step in.

**Comment by SleeplessByte** [comment]
I'm on the fence on taking this over. @tannerlinsley do you have an update to the points from this post: https://medium.com/@tannerlinsley/%EF%B8%8F-introducing-react-static-a-progressive-static-site-framework-for-react-3470d2a51ebc?

We're currently using react-static in multiple production sites, as well as gatsby. The _only_ real reason I didn't use `Next.js` was that blog post. If there's still enough difference between what we can do with react-static and next.js, I can personally pick this up (which would then be back by the clients that use this in prod). If next.js is really _good enough_ now to run a static site (pre-rendererd, and fast aka lighthouse 100), we'll also make that switch.

**Comment by digitalkaoz** [comment]
Im happy to contribute. For me react-static is the most flexible SSG out there which doesnt force me todo things in a certain manor.

**Comment by tannerlinsley** [comment]
@SleeplessByte 
- I believe that for a majority of smaller sites and users, static sites are still sufficient and work great.
- I still believe that React Static is a valuable entity in the progressive static site world and a great alternative to the heavy-handed Gatsby. 
- Next.js thus far has delivered a much better experience than it originally did, even in the static site realm, but strictly in terms of SSG, I believe other tools are better for it. If SSG is still your passion, then you'll be better off staying here. With that said, I am not moving to Next.js and expecting it to be a better SSG. I am moving to Next.js because I want to use it and study it in hopes to tackle the larger harder problems that come with sites that need to scale beyond the limits of progressive SSG frameworks.

**Comment by tannerlinsley** [comment]
Also, it never hurts to evaluate frameworks again from time to time. I have been doing that with Gatsby and Next every few months, which is what led me to this decision. If you have the time, an open mind and more information is always a good decision.

**Comment by SleeplessByte** [comment]
Let me get back to you tomorrow, but I think we can take this on fully.

**Comment by ScriptedAlchemy** [comment]
This must’ve been a hard decision. You’ve been very dedicated and RS has become very useful. I’ve seen it used for some neat things: from library documentation generation to micro frontends. I understand your reasoning. I’ve come to the decision to trial smooth-code’s implementation of Universal. I’ve been reading the codebase of next.js as well. It solves a variety of problems and works on multiple server setups. It’s an appealing project to contribute and collaborate on. RS community should thrive. I wish I had more time for OSS, always a friend to the project. You’ll make great contributions to ziet OSS efforts

**Comment by SleeplessByte** [comment]
@tannerlinsley We can take this on. We're going to keep using this for the current production site and use it for the new ones at least the next 6 months -- probably for the foreseeable future.

**Comment by arlair** [comment]
Thanks for your great work on react-static and best of luck moving forward.

I looked at next.js fairly recently and found the static route creation from code was not as easy as react-static. 

I also did not like how getInitialProps was forced onto the client side components for static pages that did not need to have it, as the data fetching and UI components were combined. I haven't looked at the latest releases though, but I preferred how react-static fetches data.

Gatsby also looks to have improved a lot since last I checked it out, but is still sounds a lot more complicated API.
